src/Tools/dotnet-monitor/CollectionRules/CollectionRuleService.cs

```diff
@@ -114,13 +152,13 @@ protected override async Task ExecuteAsync(CancellationToken token)
                 // Get a copy of the container list to avoid having to
                 // lock the entire list during stop and restart of all containers.
                 CollectionRuleContainer[] containers;
-                lock (_containers)
+                lock (_containersMap)
                 {
-                    containers = _containers.ToArray();
+                    containers = _containersMap.Values.ToArray();
```

Comment:
I don't think this is an issue but the order in which we operate on the containers won't be stable.